mirrord/operator/src/client.rs

```diff
@@ -0,0 +1,130 @@
+use std::io;
+
+use async_trait::async_trait;
+use futures::{SinkExt, StreamExt};
+use mirrord_config::{agent::AgentConfig, target::TargetConfig};
+use mirrord_kube::api::AgentManagment;
+use mirrord_progress::Progress;
+use mirrord_protocol::{ClientMessage, DaemonMessage};
+use tokio::{
+    net::{TcpStream, ToSocketAddrs},
+    sync::mpsc,
+};
+
+use crate::protocol::{AgentInitialize, OperatorCodec, OperatorRequest, OperatorResponse};
+
+pub struct OperatorApi<T: ToSocketAddrs> {
+    addr: T,
+    agent: AgentConfig,
```

Comment:
Adding to my previous comment - seems the only reason the operator API is even aware of the Agent is the ability to configure the Agent from the layer. This isn't required though. Agents should be configured at the operator level, not on a per-agent basis through the layer.